### PR TITLE
Initial arm64 support

### PR DESCRIPTION
**Pull request by mika**
Based on https://github.com/grml/grml-debootstrap/pull/191 by @GavinPacini  - rebased on top of current master to address merge conflict, squashed commits into one single commit, minor styling corrections.

Closes: https://github.com/grml/grml-debootstrap/issues/169

### CODE REVIEW

**Comment by adrelanos** [comment]
Unfortunately, a new merge conflict.

**Comment by mika** [comment]
> Unfortunately, a new merge conflict.

But easy to handle, done

**Comment by adrelanos** [comment]
https://github.com/grml/grml-debootstrap/pull/191#issuecomment-1764517044

File: grml-debootstrap [hunk 1]
```diff
@@ -1561,61 +1592,83 @@ grub_install() {
...
+        chroot "$MNTPOINT" grub-install --target=i386-efi --efi-directory=/boot/efi --uefi-secure-boot --removable "/dev/$LOOP_DISK"
+      else
+        if ! chroot "${MNTPOINT}" dpkg --list grub-efi-amd64-signed 2>/dev/null | grep -q '^ii' ; then
+          echo "Notice: grub-efi-amd64-signed package not present yet, installing it therefore."
+          # shellcheck disable=SC2086
+          DEBIAN_FRONTEND=$DEBIAN_FRONTEND chroot "$MNTPOINT" apt-get -y --no-install-recommends install $DPKG_OPTIONS grub-efi-amd64-bin grub-efi-amd64-signed
+        fi
+        chroot "$MNTPOINT" grub-install --target=x86_64-efi --efi-directory=/boot/efi --uefi-secure-boot --removable "/dev/$LOOP_DISK"
       fi
+<<<<<<< HEAD
```
Comment (adrelanos) [inline grml-debootstrap:1671]: `<<<<<<< HEAD` is probably a mistake?

File: grml-debootstrap [hunk 2]
```diff
@@ -1626,26 +1679,27 @@ if [[ -z "${GRUB}" ]] || ! dd if="${GRUB}" bs=512 count=1 2>/dev/null | cat -v |
       fi
       chroot "$MNTPOINT" grub-install --target=x86_64-efi --efi-directory=/boot/efi --uefi-secure-boot --removable "/dev/$LOOP_DISK"
       chroot "$MNTPOINT" grub-install --target=i386-pc "/dev/$LOOP_DISK"
+=======
```
Comment (adrelanos) [inline grml-debootstrap:1682]: This too.

File: grml-debootstrap [hunk 3]
```diff
@@ -1626,26 +1679,27 @@ if [[ -z "${GRUB}" ]] || ! dd if="${GRUB}" bs=512 count=1 2>/dev/null | cat -v |
...
+          ;;
+        # NOTE - we might need to distinguish between further filesystems
+        *)
+          chroot "${MNTPOINT}" grub-mkimage -O i386-pc -p "(hd0,msdos1)/boot/grub" -o /tmp/core.img biosdisk part_msdos ext2
+          ;;
+      esac
+
+      dd if="${MNTPOINT}/tmp/core.img" of="${ORIG_TARGET}" conv=notrunc seek=1
+      rm -f "${MNTPOINT}/tmp/core.img"
+>>>>>>> ecddb6f (Initial arm64 support)
```
Comment (adrelanos) [inline grml-debootstrap:1700]: Here too.

File: grml-debootstrap (hunk 1 above)
Comment (mika) [reply grml-debootstrap]: Uhm yeah, sorry, pushed the correct version now

**Comment by mika** [comment]
I *think* `grub_install()` could be simplified now with f4d3906aae588d8dac1a146f05151211073cbb80 being present?

**Comment by adrelanos** [comment]
670035bdde9ac232b3869eea4a66f5412d1d1132 build + boot success.

**Comment by mika** [comment]
> ```
> git checkout 670035bdde9ac232b3869eea4a66f5412d1d1132
> sudo make install
> ```
> 
> [670035b](https://github.com/grml/grml-debootstrap/commit/670035bdde9ac232b3869eea4a66f5412d1d1132) build + boot success.

Great, thanks for testing, @adrelanos! :) Is @GavinPacini somehow connected/related to you/your work? I'm wondering whether to accept this PR which _I_ did with some commit squashing and style adjustments but under _Gavin's_ name is a) fine for him, and b) whether we could further improve current state? (Also making sure that we didn't break non-arm64 would be good :))

**Comment by mika** [comment]
> I cannot speak for @GavinPacini.
> 
> > Is @GavinPacini somehow connected/related to you/your work?
> 
> That shouldn't matter. I hope it doesn't matter. I didn't think any transparency is required here. But for full transparency...
> 
> Gavin has been active here: https://forums.whonix.org/t/whonix-on-mac-m1-arm-user-support-still-unsupported-at-time-of-writing/11310
> 
> At some point I said something like "the first step to make Whonix on ARM64 work would be contributing #169 to upstream grml-debootstrap". Gavin contributed towards that.

Ah no worries, I just wanted to know whether you could reach out to Gavin since he didn't respond in the last few days. :)

> The pull request was made from grml-debootstrap (upstream) perspective. So upstream doesn't need to spend time on any unrelated history.
> 
> > I'm wondering whether to accept this PR which I did with some commit squashing and style adjustments but under Gavin's name is a) fine for him,
> 
> So the easy, proper attribution so we don't have to wait for Gavin or anyone's input would be... Merge Gavin's pull request "somehow" if that's still possible. Quick and dirty. Then in the next commit "revert" to your [670035b](https://github.com/grml/grml-debootstrap/commit/670035bdde9ac232b3869eea4a66f5412d1d1132).
> 
> Otherwise attribution in the commit message should be sufficient too?
> 
> When I look at https://github.com/grml/grml-debootstrap/pull/214/commits it reads:
> 
> > @[**GavinPacini**](https://github.com/grml/grml-debootstrap/commits?author=GavinPacini) authored and @[mika](https://github.com/grml/grml-debootstrap/commits?author=mika) committed.
> 
> So it seems that would already as of now been properly recorded in the git history?

Absolutely, the author would be Gavin, and I'm the committer only, would be fine for *me*

> > b) whether we could further improve current state?
> 
> I hope that could be done in form of new issues posted and new pull requests?

Also fine for me :)

> > (Also making sure that we didn't break non-arm64 would be good :))
> 
> I can test booworm amd64 now.

I'd certainly appreciate that, once we know we're good I'd merge this PR and we can continue work from there then :)

thx!

**Comment by mika** [comment]
> Bookworm amd64 build + boot success.
> 
> ```
> qemu-system-x86_64 \
>     -m 1024 \
>     -drive format=raw,file=/home/user/grml-debootstraptestbin/test.img
> ```

Ok great, thanks for testing!

> > Ah no worries, I just wanted to know whether you could reach out to Gavin since he didn't respond in the last few days. :)
> 
> Unfortunately, I don't have any other contact.

No worries, then I suggest we wait until next week, and if we don't have any objections until then from @GavinPacini I'd say we merge this PR. :)

**Comment by mika** [comment]
Ok, merged now! :) Thanks everyone!

**Comment by GavinPacini** [comment]
Great to see that this was completed and merged. Thanks @adrelanos and @mika for getting this over the line!

I have not been as active on open-source contributions lately, apologies, but I do appreciate your support in finishing this. The growing arm64 community will be very happy I'm sure.

**Comment by mika** [comment]
@GavinPacini wonderful, thanks for reporting back! :)

**Pull request merged**